### ISSUE TITLE
GPD Win4Pro: ChimeraOS can't detect Lexar NVME SSD 

### ISSUE THREAD

**Issue by parth-menon** [ChimeraOS/chimeraos#775]
This issue occurs both during install and when booting chimera from SSD. 
ChimeraOS can't initialize the internal nvme SSD
My internal SSD details:
Model: Lexar SSD ARES 4TB
VendorId: 0x1d97

This is the exact same issue that i have:
https://www.linux.org/threads/lexar-nm790-nvme-fails-to-initialize.46315/page-2

The same issue is there on other Linux distros as well.

After installing ChimeraOS on an external SSD, which also can't detect the internal ssd, I created a patch with this change 
https://bugzilla.kernel.org/attachment.cgi?id=305061&action=diff

And then it starts working.

**Comment by pastaq** [comment]
Can you submit the patch https://github.com/ChimeraOS/linux-chimeraos ?

**Comment by parth-menon** [comment]
Yes definitely. I'll just do some more testing and then upload the patch with a test plan.

Thanks!

> Can you submit the patch https://github.com/ChimeraOS/linux-chimeraos ?

**Comment by parth-menon** [comment]
> I believe this is fixed in linux kernel v6.5.5. Since that is already in the testing channel, could you test that and report here?

Yes! It's working on unstable branch with 6.5.5 kernel without my patch.

**Comment by pastaq** [comment]
Cool, thanks for testing. We'll mark as closed when we push to stable on that version.

**Comment by parth-menon** [comment]
@pastaq @BoukeHaarsma23 sorry to report that 6.5.5 is not working with this issue.
It was only booting because i had an external ssd, also with ChimeraOS and my patch, which was loading the internal ssd frzr-root.

For now I'll create the patch and upload it as a PR

**Comment by parth-menon** [comment]
I tested this again a couple of times.
It looks like the issue is fixed with 6.6.0 kernel but the installer still can't see the nvme during installation.

@pastaq

**Comment by parth-menon** [comment]
> Let's update the installer once 6.6 is out. That is expected next week

Perfect. Thanks!

**Comment by parth-menon** [comment]
> Sorry for replying this late: I have updated the installer with a new kernel. Please let me know if the issue is solved now.

Perfect, thank you. 
I'll try it out.

**Comment by vithusel** [comment]
Confirming it is now working and the installer sees the drive (I installed the os for the first time last night) 

I did experience a odd drop in connection which broke the OS without causing a kernel panic. Nothing got captured in the logs annoyingly. Will keep an eye on it. 

It specifically happened when I used rpcs3 to perform precaching on my games which is IO intensive. In any case. It's a kernel issue and not OS so I think this can be closed
